### PR TITLE
I47 add annotations

### PR DESCRIPTION
**Pull request by allishultes**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      
Issue #47 

**Describe what the PR does**    

- [x] Adds CRUD operations for annotations in the paper edit view

Details (from #47 ):
- [x] Users can select text and then select "highlight" to add a label to their transcript (see screenshot below). Please check this is also reflected properly in the database. 
- [x] The highlights appear with the proper colour / name of the label that is selected when the highlight is made.
- [x] CRUD operations are reflected in both the database and in the TranscriptContainer view.*
  - [x] You can create new annotations
  - [x] You can delete annotations
  - [x] You can edit annotations

**State whether the PR is ready for review or whether it needs extra work**    

**Additional context**    

For the reviewer:

I added a field to the label in `labels` props for determining which one was active, i.e., had been last selected by a user. It might be better to just assign a separate prop for keeping track of the active label id? 

There seems to be quite a lot of rendering going on... not sure if it's my implementation or just a lot of state changes to keep track of. 

The design of the annotation pop-up is ugly AF. For now I'm assuming that's out of scope but lmk if I should consider that part of the scope of this ticket.

### CODE REVIEW

**Comment by emettely** [comment]
I've verified:
- [x] deleting
- [x] adding
- [x] editing
- [x] updating a label updates all the associated annotations
- [ ] removing all labels from the project

The last one throws an error:
Screenshot 2020-04-24 at 17 31 52


TBD in a new issue:
- We need to add default highlights. 
Screenshot 2020-04-24 at 17 30 12
- when the associated label is removed from annotations, the label is replaced with "default"

File: src/Components/PaperEditor/TranscriptsContainer/TranscriptTabContent/TranscriptMenu.js [hunk 1]
```diff
@@ -13,16 +13,27 @@ import { faHighlighter, faCog } from '@fortawesome/free-solid-svg-icons';
...
+    });
+    if (previousActiveLabel) {
+      previousActiveLabel.active = false;
+    }
+    const activeLabel = tempLabels.find((label) => {
+      return label.id === labelId;
+    });
+    activeLabel.active = true;
+    props.setLabels(tempLabels);
   };
```
Comment (emettely) [inline src/Components/PaperEditor/TranscriptsContainer/TranscriptTabContent/TranscriptMenu.js:30]: Here you are adding a logic to set the parent components state. In a child component you shouldn't be dealing with how the parent component's state is set. You should move this code up to the parent component, and not pass `setLabels` at all. 

e.g. line 69 of your code
```
<Row ... onClick={ (e) => props.updateSelectedLabel(e, label.id) }>
```

File: src/Components/PaperEditor/TranscriptsContainer/index.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ const TranscriptsContainer = ({
         <TranscriptTabContent
           projectId={ projectId }
           transcriptId={ id }
-          labels={ labels }
+          labels={ labelsOptions }
```
Comment (emettely) [inline src/Components/PaperEditor/TranscriptsContainer/index.js:55]: rename all the `labelsOptions` back to `labels`

File: src/Components/PaperEditor/index.js [hunk 3]
```diff
@@ -60,20 +72,56 @@ const PaperEditor = (props) => {
...
               return { ...doc.data(), id: doc.id, display: true };
             })
           );
         });
       } catch (error) {
-        console.error('Error getting labels: ', error);
+        console.error('Error getting annotations: ', error);
       }
     };
+    if (!transcripts) {
```
Comment (emettely) [inline src/Components/PaperEditor/index.js:117]: probably worth setting annotations to `null` to begin with, so you can use annotations instead to check if you're fetching for annotations. This way you also don't have to subscribe to transcripts in the useEffect.

File: src/Components/PaperEditor/index.js [hunk 4]
```diff
@@ -60,20 +72,56 @@ const PaperEditor = (props) => {
       }
     };
 
+    if (!transcripts) {
```
Comment (emettely) [inline src/Components/PaperEditor/index.js:75]: use paperEdits instead.

File: src/Components/PaperEditor/index.js [hunk 5]
```diff
@@ -51,6 +55,14 @@ const PaperEditor = (props) => {
...
```
Comment (emettely) [inline src/Components/PaperEditor/index.js:58]: use projects instead

File: src/Components/PaperEditor/TranscriptsContainer/index.js (hunk 2 above)
Comment (emettely) [reply src/Components/PaperEditor/TranscriptsContainer/index.js:55]: there are other areas that labelsOptions, I didn't mark all of them so please find them :)

File: src/Components/PaperEditor/index.js (hunk 3 above)
Comment (allishultes) [reply src/Components/PaperEditor/index.js:117]: Changing this causes the view not to load; the original code was also written in a way where transcripts were checked for each of these values. I'm not sure why this happens and your change makes sense ... but it causes the view to try to load before the other props have been grabbed from Firebase.

**Pull request merged**